docs/llm/antipatterns.adoc

```diff
@@ -0,0 +1,703 @@
+
+* Silent failures
+* Content mysteriously missing from output
+* Hard to debug
+
+**Current behavior:** Parser returns null on failure (logged but not thrown)
+
+**✅ Better approach (future):**
+
+* Throw `ParsingException` with details
+* Oven collects errors for summary
+* User gets clear error message
+
+**Code example:**
+
+```java
+❌ Current (anti-pattern):
+public DocumentModel processFile(File file) {
+  try {
+    // ... parsing
+  } catch (Exception e) {
+    LOGGER.error("Failed to parse", e);
+    return null;  // Silent failure!
+  }
+}
+
+✅ Better:
+public DocumentModel processFile(File file) throws ParsingException {
+  try {
+    // ... parsing
+  } catch (Exception e) {
+    throw new ParsingException(file, e);  // Explicit failure
+  }
+}
+```
+
+**Status:** Technical debt; discussed in Issue #234
+
+---
+
+=== ❌ Antipattern: Overriding parse() Instead of Hooks
+
+**Why wrong:**
+
+* `parse()` is template method, controls flow
+* Overriding breaks header/body separation
+* Duplicates base class logic
+
+**✅ Correct approach:**
+
+* Override `processHeader()` and `processBody()` hooks
+* Let base class handle flow control
+
+**Code example:**
+
+```java
+❌ Wrong:
+public class MyEngine extends MarkupEngine {
+  @Override
+  public DocumentModel parse(JBakeConfiguration config, File file) {
+    // Reimplements entire flow - bad!
+    // ... duplicate logic
+  }
+}
+
+✅ Correct:
+public class MyEngine extends MarkupEngine {
+  @Override
+  protected void processHeader(ParserContext context) {
+    // Just handle header extraction
+  }
+  
+  @Override
+  protected void processBody(ParserContext context) {
+    // Just handle body conversion
+  }
+}
+```
+
+**Related:** `MarkupEngine.java` template method pattern
+
+---
+
+== Content Authoring Anti-Patterns
+
+=== ❌ Antipattern: Omitting Required Metadata
+
+**Why wrong:**
+
+* Content won't render (missing type or status)
+* Default status is "draft" (not rendered)
+* Confusing "where's my content?" issues
+
+**✅ Correct approach:**
+
+* Always specify `type` and `status` in header
+* Or configure defaults in `jbake.properties`
+
+**Code example:**
+
+```markdown
+❌ Wrong:
+title=My Post
+date=2025-11-22
+~~~~~~
+Content here...
+# Missing type and status!
+
+✅ Correct:
+title=My Post
+date=2025-11-22
+type=post
+status=published
+~~~~~~
+Content here...
+
+# Or configure defaults:
+# default.type=post
+# default.status=published
+```
+
+---
+
+=== ❌ Antipattern: Inconsistent Date Formats
+
+**Why wrong:**
+
+* Parsing errors
+* Dates not sortable
+* Archive generation fails
+
+**✅ Correct approach:**
+
+* Use consistent format matching `date.format` config
+* Default: `yyyy-MM-dd`
+* Stick to ISO 8601 formats
+
+**Code example:**
+
+```markdown
+❌ Wrong:
+date=November 22, 2025  # Doesn't match default format!
+date=22/11/2025         # Ambiguous!
+
+✅ Correct:
+date=2025-11-22         # ISO 8601, matches default
+
+# Or configure format:
+# date.format=dd-MM-yyyy
```

Comment:
This is a big problem related to `java.util.Date`. We need to think of a date parser which favours `java.time` related methods and only falls back on `java.util.Date` on special occasions (e.g. fallback mode, legacy mode, etc.)